File: app/src/main/java/com/skyd/imomoe/util/dlna/dmc/ILogger.java
package com.skyd.imomoe.util.dlna.dmc;

import android.text.TextUtils;
import android.util.Log;

import com.skyd.imomoe.BuildConfig;

public interface ILogger {
    String PREFIX_TAG = "DLNACast_";

    void v(String msg);

    void d(String msg);

    void i(String msg);

    void w(String msg);

    void e(String msg);

    class DefaultLoggerImpl implements ILogger {
        private final String TAG;
        private final boolean DEBUG;

        public DefaultLoggerImpl(Object object) {
            this(object, BuildConfig.DEBUG);
        }

        public DefaultLoggerImpl(Object object, boolean debug) {
            String className = object.getClass().getSimpleName();
            if (TextUtils.isEmpty(className)) {
                if (object.getClass().getSuperclass() != null) {
                    className = object.getClass().getSuperclass().getSimpleName();
                } else {
                    className = "$1";
                }
            }
            TAG = PREFIX_TAG + className;
            DEBUG = debug;
        }

        @Override
        public void v(String msg) {
            if (DEBUG) {
                Log.v(TAG, msg);
            }
        }

        @Override
        public void d(String msg) {
            if (DEBUG) {
                Log.d(TAG, msg);
            }
        }

        @Override
        public void i(String msg) {
            if (DEBUG) {
                Log.i(TAG, msg);
            }
        }

        @Override
        public void w(String msg) {
            if (DEBUG) {
                Log.w(TAG, msg);
            }
        }

        @Override
        public void e(String msg) {
            if (DEBUG) {
                Log.e(TAG, msg);
            }
        }
    }
}

File: app/src/main/java/com/skyd/imomoe/util/dlna/dmc/ILogger.kt
package com.skyd.imomoe.util.dlna.dmc

import android.util.Log
import com.skyd.imomoe.BuildConfig
import com.skyd.imomoe.util.*

interface ILogger {
    fun v(msg: String?)
    fun d(msg: String?)
    fun i(msg: String?)
    fun w(msg: String?)
    fun e(msg: String?)

    class DefaultLoggerImpl @JvmOverloads constructor(
        `object`: Any,
        debug: Boolean = BuildConfig.DEBUG
    ) : ILogger {
        private val TAG: String
        private val DEBUG: Boolean

        override fun v(msg: String?) {
            if (DEBUG) logV(TAG, msg.toString())
        }

        override fun d(msg: String?) {
            if (DEBUG) logD(TAG, msg.toString())
        }

        override fun i(msg: String?) {
            if (DEBUG) logI(TAG, msg.toString())
        }

        override fun w(msg: String?) {
            if (DEBUG) logW(TAG, msg.toString())
        }

        override fun e(msg: String?) {
            if (DEBUG) logE(TAG, msg.toString())
        }

        init {
            var className = `object`.javaClass.simpleName
            if (className.isNullOrEmpty()) {
                className = if (`object`.javaClass.superclass != null) {
                    `object`.javaClass.superclass.simpleName
                } else {
                    "$1"
                }
            }
            TAG = PREFIX_TAG + className
            DEBUG = debug
        }
    }

    companion object {
        const val PREFIX_TAG = "DLNACast_"
    }
}